src/inventory_ui.cpp

```diff
@@ -840,6 +840,12 @@ item_location game::inv_map_splice( item_filter filter, const std::string &title
     return inv_map_splice( filter, filter, filter, title, radius );
 }
 
+void clear_cats( std::vector<item_category *> &cats ) {
```

Comment:
> Alternatively, you can use a std::list to store the categories

I did successfully stub out an implementation using this approach but never got around to completing it.